Media Controls: Volume slider should always show if we have enough room

rr-codes opened the pull request:
#### e4772fad1ca24d48d3d2fd2bcfd313fd6c4dcccf
<pre>
Media Controls: Volume slider should always show if we have enough room
<a href="https://bugs.webkit.org/show_bug.cgi?id=259721">https://bugs.webkit.org/show_bug.cgi?id=259721</a>
rdar://113243353

Reviewed by Dean Jackson.

On visionOS, the volume slider should always show within inline video, if there is enough room.

* Source/WebCore/DerivedSources-input.xcfilelist:
* Source/WebCore/DerivedSources.make:
* Source/WebCore/Modules/modern-media-controls/controls/vision-media-controls.css:
(.media-controls.vision &gt; .controls-bar.top-right,):
(.media-controls.vision &gt; .controls-bar.top-right):
(.media-controls.vision .controls-bar.simple-layout .volume-container &gt; .slider.vision.volume,):
(.media-controls.vision .controls-bar.simple-layout.top-right &gt; .overflow):
(.media-controls.vision .volume-container):
(.media-controls.vision .volume-container &gt; .mute.bar):
(.media-controls.vision:not(.audio) .volume-container &gt; .background-tint &gt; .blur,):
(.media-controls.vision.fullscreen &gt; .controls-bar.top-right,): Deleted.
(.media-controls.vision.fullscreen &gt; .controls-bar.top-right): Deleted.
(.media-controls.vision .controls-bar.simple-layout &gt; *:not(.background-tint)): Deleted.
(.media-controls.vision:not(.audio) button &gt; .background-tint &gt; .blur): Deleted.
* Source/WebCore/Modules/modern-media-controls/controls/vision-media-controls.js:
(VisionMediaControls.prototype._createControls):
(VisionMediaControls.prototype._performVolumeContainerLayout):
(VisionMediaControls.prototype._performTopRightControlsBarLayout):
(VisionInlineMediaControls.prototype._createControls):
(VisionInlineMediaControls.prototype._performLayout):
(VisionInlineMediaControls.prototype._topRightControlsBarChildren):
(VisionInlineMediaControls.prototype._performVolumeContainerLayout):
(VisionFullscreenMediaControls.prototype._createControls):
(VisionFullscreenMediaControls.prototype._performVolumeContainerLayout):
(VisionFullscreenMediaControls.prototype._topRightControlsBarChildren):
* Source/WebCore/Modules/modern-media-controls/controls/vision-volume-container.js: Added.
(VolumeContainer.prototype.get children):
(VolumeContainer.prototype.set children):
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:

Canonical link: <a href="https://commits.webkit.org/266566@main">https://commits.webkit.org/266566@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/c74967d02d22077bf3191e5388515989c22c8f4a

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/14142 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/14458 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/14795 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/15882 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/13407 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/14246 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/16969 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/14540 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/16081 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/14316 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/14893 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/11992 "Passed tests") | [❌ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/16597 "") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/12176 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/12753 "Passed tests") | [❌ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/2/builds/16597 "") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/13255 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/12917 "Passed tests") | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/2/builds/16597 "") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/13463 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/11325 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/12742 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/3429 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/17077 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/13306 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

rr-codes commented on Source/WebCore/Modules/modern-media-controls/controls/vision-media-controls.js:
```diff
@@ -36,6 +36,30 @@ const MaxWidthForBottomControlsBarInFullscreen = 602;
 const MinWidthToShowSeekingControlsForAudio = 120;
 const MinWidthToShowTimeControlForAudio = 262;
 
+class VolumeContainer extends LayoutNode
```
move to its own file

rr-codes commented on Source/WebCore/Modules/modern-media-controls/controls/vision-media-controls.js:
```diff
@@ -242,12 +266,19 @@ class VisionInlineMediaControls extends VisionMediaControls
     {
         super._createControls();
+        
+        this.volumeSlider = new VisionSlider(this, "volume");
```
only show slider if there's room

rr-codes commented on Source/WebCore/Modules/modern-media-controls/controls/vision-media-controls.js:
```diff
@@ -242,12 +266,19 @@ class VisionInlineMediaControls extends VisionMediaControls
 
+        this.volumeContainer = new VolumeContainer("volume-container");
+        this.volumeContainer.children = [this.volumeSlider, this.muteButton];
+        this.volumeContainer.hasBackgroundTint = true;
```
is this line even needed

rr-codes commented on Source/WebCore/Modules/modern-media-controls/controls/vision-media-controls.css:
```diff
@@ -76,12 +94,12 @@
 
-.media-controls.vision.fullscreen > .controls-bar.top-right,
-.media-controls.vision.fullscreen > .controls-bar.top-right * {
+.media-controls.vision > .controls-bar.top-right,
```
move out of fullscreen section

pxlcoder commented on Source/WebCore/WebCore.xcodeproj/project.pbxproj:
```diff
@@ -3,7 +3,7 @@
 	classes = {
 	};
-	objectVersion = 51;
+	objectVersion = 54;
```
Probably don't want to commit this.

pxlcoder commented on Source/WebCore/DerivedSources.make:
```diff
@@ -1984,6 +1984,7 @@ MODERN_MEDIA_CONTROLS_SCRIPTS = \
     $(WebCore)/Modules/modern-media-controls/controls/invalid-placard.js \
     $(WebCore)/Modules/modern-media-controls/controls/pip-placard.js \
     $(WebCore)/Modules/modern-media-controls/controls/vision-slider.js \
+	$(WebCore)/Modules/modern-media-controls/controls/vision-volume-container.js \
```
Indentation.

rr-codes committed:
Media Controls: Volume slider should always show if we have enough room
https://bugs.webkit.org/show_bug.cgi?id=259721
rdar://113243353

Reviewed by Dean Jackson.

On visionOS, the volume slider should always show within inline video, if there is enough room.

* Source/WebCore/DerivedSources-input.xcfilelist:
* Source/WebCore/DerivedSources.make:
* Source/WebCore/Modules/modern-media-controls/controls/vision-media-controls.css:
(.media-controls.vision > .controls-bar.top-right,):
(.media-controls.vision > .controls-bar.top-right):
(.media-controls.vision .controls-bar.simple-layout .volume-container > .slider.vision.volume,):
(.media-controls.vision .controls-bar.simple-layout.top-right > .overflow):
(.media-controls.vision .volume-container):
(.media-controls.vision .volume-container > .mute.bar):
(.media-controls.vision:not(.audio) .volume-container > .background-tint > .blur,):
(.media-controls.vision.fullscreen > .controls-bar.top-right,): Deleted.
(.media-controls.vision.fullscreen > .controls-bar.top-right): Deleted.
(.media-controls.vision .controls-bar.simple-layout > *:not(.background-tint)): Deleted.
(.media-controls.vision:not(.audio) button > .background-tint > .blur): Deleted.
* Source/WebCore/Modules/modern-media-controls/controls/vision-media-controls.js:
(VisionMediaControls.prototype._createControls):
(VisionMediaControls.prototype._performVolumeContainerLayout):
(VisionMediaControls.prototype._performTopRightControlsBarLayout):
(VisionInlineMediaControls.prototype._createControls):
(VisionInlineMediaControls.prototype._performLayout):
(VisionInlineMediaControls.prototype._topRightControlsBarChildren):
(VisionInlineMediaControls.prototype._performVolumeContainerLayout):
(VisionFullscreenMediaControls.prototype._createControls):
(VisionFullscreenMediaControls.prototype._performVolumeContainerLayout):
(VisionFullscreenMediaControls.prototype._topRightControlsBarChildren):
* Source/WebCore/Modules/modern-media-controls/controls/vision-volume-container.js: Added.
(VolumeContainer.prototype.get children):
(VolumeContainer.prototype.set children):
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:

Canonical link: https://commits.webkit.org/266566@main
